extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/DeviceHandler.java

```diff
@@ -0,0 +1,928 @@
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link DeviceHandler} is responsible to handling the configuration, load supported channels of an
+ * digitalSTROM device and handling commands which are send to one of the channels. <br>
+ * <br>
+ * For that it uses the {@link BridgeHandler} to execute the actual command and implements the
+ * {@link DeviceStatusListener} to get informed about changes from the accompanying {@link Device}.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ *
+ */
+public class DeviceHandler extends BaseThingHandler implements DeviceStatusListener {
+
+    private Logger logger = LoggerFactory.getLogger(DeviceHandler.class);
+
+    /**
+     * The {@link DsThingTypeProvider} add the supported thing types
+     */
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.newHashSet();
+
+    private String dSID = null;
+
+    private Device device;
+
+    private BridgeHandler dssBridgeHandler;
+
+    private Command lastComand = null;
+
+    private String currentChannel = null;
+
+    public DeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing DigitalSTROM Device handler.");
+        String configdSID = getConfig().get(DigitalSTROMBindingConstants.DEVICE_DSID).toString();
+        if (!configdSID.isEmpty()) {
+            dSID = configdSID;
+            if (getDssBridgeHandler() == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_MISSING_ERROR, "Bridge is missig");
+            } else {
+                bridgeHandlerInitialized(dssBridgeHandler, this.getBridge());
+            }
+
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "dSID is missig");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposes. Unregistering listener.");
+        if (dSID != null) {
+            if (dssBridgeHandler != null) {
+                dssBridgeHandler.unregisterDeviceStatusListener(this);
+            }
+            device = null;
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        if (!getThing().getStatus().equals(ThingStatus.UNINITIALIZED) && getDssBridgeHandler() != null) {
+            this.dssBridgeHandler.childThingRemoved(dSID);
+        }
+        updateStatus(ThingStatus.REMOVED);
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        // dispose();
```

Comment:
did you test the logic in here? Remove the commented out lines if you are sure that you don't need them